R/setup.R

```diff
@@ -1,43 +1,102 @@
 #' Setup deployment for Travis CI
 #'
 #' Creates a public-private key pair,
-#' adds the public key to the GitHub repository via [github_add_key()],
+#' adds the public key to the GitHub repository via `github_add_key()`,
 #' and stores the private key as an encrypted environment variable in Travis CI
 #' via [travis_set_var()],
 #' possibly in a different repository.
 #' The \pkg{tic} companion package contains facilities for installing such a key
 #' during a Travis CI build.
 #'
-#' @inheritParams github_add_key
 #' @inheritParams travis_repo_info
-#' @param travis_repo `[string]`\cr
+
+#' @param path `[string]` \cr
+#'   The path to the repository.
+#' @param user `[string]` \cr
+#'   Name of the Github user account.
+#' @param repo `[string]`\cr
 #'   The Travis CI repository to add the private key to, default: `repo`
 #'   (the GitHub repo to which the public deploy key is added).
+#' @template endpoint
 #'
 #' @export
-use_travis_deploy <- function(path = usethis::proj_get(), info = github_info(path),
-                              repo = github_repo(info = info), travis_repo = repo) {
+use_travis_deploy <- function(path = usethis::proj_get(),
+                              user = github_user()$login,
+                              repo = github_info()$name,
+                              endpoint = NULL) {
+
+  if (is.null(endpoint)) {
```

Comment:
I wonder if we can move this to the `endpoint()` function here and everywhere.

Reply:
Done. Default is now `endpoint = get_endpoint()` and all the if-clauses are removed